ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/RestoreCommandsCompactor.java

```diff
@@ -24,12 +24,17 @@
 import java.util.Map;
 import java.util.Optional;
 import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Util for compacting the restore commands
  */
 public final class RestoreCommandsCompactor {
 
+  static QueryId greatestQueryId;
```

Comment:
making this static is a design smell. if we need to access information from within this class we have two options (1) we can make `compact` return an object that is a pair of `greatestQueryId` and the `List<QueuedCommand>` (2) we can make `RestoreCommandsCompactor` non-static, and create the instance where we need it

you can read along https://stackoverflow.com/questions/7026507/why-are-static-variables-considered-evil for more reasons why